pkg/operator/ceph/object/spec.go

```diff
@@ -1130,3 +1151,28 @@ func GetHostnameFromEndpoint(endpoint string) (string, error) {
 
 	return hostname, nil
 }
+
+func (c *clusterConfig) isLocalizedReadAffinitySet() bool {
+	if c.store.Spec.Gateway.RgwReadAffinity != nil && c.store.Spec.Gateway.RgwReadAffinity.Type == localizedReadReplicaPolicy {
```

Comment:
I realize that technically we are being a smart by only setting up the script stuff when the policy is `localize`, but I would suggest that we instead just check if the user has a non-nil read affinity spec. This will make it easier to support other read affinity types in the future without having to change rook internal code -- only the kubebuilder enums, which is something a user could patch themselves with a workaround.

```suggestion
func (c *clusterConfig) isReadAffinitySet() bool {
	// if user specifies read affinity options, assume we need read affinity setup
	if c.store.Spec.Gateway.RgwReadAffinity != nil {
```